### DISCUSSION TITLE
About "No more implicit connection to the default namespace"

### DISCUSSION THREAD

**Discussion by hnxzc** [socketio/socket.io#4037] [Q&A] [answered]
How to connect to the main namespace or other custom namespaces without "socket.io-client"?

I am developing a wechat micro program, any third party framework is not supported.

Before 3.x, I can connect socket.io server like this 
`wx.connectSocket({
      url: 'ws://127.0.0.1:3000/socket.io/?EIO=3&transport=websocket'
    })`

But now it does not work, connection will always be closed after 45000ms, because I cannot successfully joined a namespace before the "connectTimeout" expired.

**Comment by darrachequesne** [comment, marked as answer]
You need to send a CONNECT packet: `40`.

4 => Engine.IO "message" type
0 => Socket.IO CONNECT type

Edit: or `40/my-custom-namespace,`, for a custom namespace

More information here: https://github.com/socketio/socket.io-protocol

**Reply by hnxzc** [reply]
Thanks!!!
